src/charm.py

```diff
@@ -78,6 +84,19 @@ def unit_peer_data(self) -> MutableMapping[str, str]:
 
         return self.peer_relation.data[self.unit]
 
+    def _on_storage_attached(self, event: StorageAttachedEvent) -> None:
+        path = event.storage.location if event.storage else None
+        if not path:
+            logger.error("Unable to find storage in StorageAttachedEvent")
+            event.defer()
+            return
+
+        self.unit_peer_data.update({"logs": "attached"})
```

Comment:
Do this trigger a new rolling restart of Kafka ? I suppose as we add storage during operations, we should trigger here a restart to provide the added directories to the properties file, right?

Reply:
That will be addressed in future PRs. This is for basic functionality at this stage.